### PR TITLE
feat: update major dependencies

### PR DESCRIPTION
**Pull request by ludtkemorgan**
This PR addresses #1591 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This upgrades several dependencies that have already been upgraded in core this includes:
- Node from 14 to 18
- Next to 12
- React to 18
- Cypress from 9 to 12

It also accomplishes the following:
- removes storybook from detroit-ui-components
- update all test ids to data-testid

## How Can This Be Tested/Reviewed?

This touches every page so it is recommended to go to every page and do a quick check. These changes are already in and have been tested so we don't have to be as thorough, but there are some changes that are unique so a quick passover is recommended. 

You will need to test locally since it has to have a new deployed version of the backend

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/partners/src/components/listings/PaperListingForm/UnitsSummaryAmiForm.tsx [hunk 1]
```diff
@@ -74,6 +74,12 @@ const UnitsSummaryAmiForm = ({
     [amiInfo, setAmiPercentageOptions]
   )
 
+  useEffect(() => {
```
Comment (ludtkemorgan) [inline sites/partners/src/components/listings/PaperListingForm/UnitsSummaryAmiForm.tsx:77]: There is an edge case only appearing on the cypress tests where the amiChartId is not being updated before the onChange event happens. By moving it to the useEffect we can guarantee that the order will happen correctly

File: sites/partners/src/components/users/FormSignInAddPhone.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import React from "react"
```
Comment (ludtkemorgan) [inline sites/partners/src/components/users/FormSignInAddPhone.tsx:1]: This file and the ones below it were accidentally missed when pulling over the refactoring commits from core. Specifically where we pulled things out of ui-components and into the corresponding packages they are used in

**Comment by emilyjablonski** [comment]
Screenshot 2023-04-14 at 4 39 32 PM
Looking great! This is the only issue I've seen so far in that right hand panel on the housings basics page.

**Comment by ludtkemorgan** [comment]
> Screenshot 2023-04-14 at 4 39 32 PM Looking great! This is the only issue I've seen so far in that right hand panel on the housings basics page.

Nice catch, there were also a couple of places where the markdown wasn't getting formatted. I've pushed up an update (remove new line below the RenderIf)

**Review by emilyjablonski — APPROVED** [review]
LGTM! Down to bash next week if we have time 🔌

**Comment by ColinBuyck** [comment]
Little thing but noticing that the "page not found" page is different. I definitely like the PR version better but just want to flag to make sure that its not pointing to any other issues
Screenshot 2023-04-18 at 4 52 52 PM

**Comment by ColinBuyck** [comment]
Also noticing when trying to add a new listing, I get an alert to resolve errors but don't see anything marked. Additionally, I get the following error in the console.
Screenshot 2023-04-18 at 5 19 33 PM

**Comment by emilyjablonski** [comment]
^ spooky to me that cypress didn't see this

**Comment by ColinBuyck** [comment]
Good point 👻 @emilyjablonski  I revisited this after deleting node modules and reinstalling everything, and can't seem to reproduce the failed listing publish from before but without fail, the console errors show just from opening the add listing page.

**Comment by ludtkemorgan** [comment]
@ColinBuyck That console error is happening core as well. I created a ticket for it recently. https://github.com/bloom-housing/bloom/issues/3392

**Comment by ludtkemorgan** [comment]
> Little thing but noticing that the "page not found" page is different. I definitely like the PR version better but just want to flag to make sure that its not pointing to any other issues Screenshot 2023-04-18 at 4 52 52 PM

Nice catch. This was because of a new error page that I added. I've updated it to the following. It's different than what is in prod, but I think this is what was originally intended
image

**Review by ColinBuyck — APPROVED** [review]
LGTM assuming Cypress test are just being being flakyyy 🫓
